_posts/2022-08-02-stimzi-based-infrastructure-as-a-java-code.md

```diff
@@ -0,0 +1,284 @@
+---
+layout: post
+title: "Strimzi based Apache Kafka infrastructure as a Java code"
+date: 2022-08-01
+author: jakub_scholz
+---
+
+When you say _Infrastructure as Code (IaC)_, people usually imagine tools such as Terraform, Ansible or Amazon AWS CloudFormation.
+All these tools use some kind of configuration or markup language to write the code.
+Terraform uses the HCL language, Ansible uses YAML and AWS CLoudFormation can use either YAML or JSON.
+Strimzi supports this pattern as well.
+You describe your Strimzi based Apache Kafka infrastructure in the YAML format when you create the Kubernetes custom resources.
+And Kubernetes together with the Strimzi operators take care of transforming this _code_ into the actual Apache Kafka brokers, users, topics or for example connectors.
+But what if you want to use a full-fledged programming language for your infrastructure ... for example Java?
```

Comment:
Should we motivate more (here in the lede) why you might want to do it with Java? Or at least say we'll answer the "Why" question in the main content?

Reply:
I went for the second option you suggested and reworded it to make it more clear the _why_ is covered in the blog post as well. 

TBH, I think this is a bit weird. We typically use the lede in the blog posts. But we do not use it anywhere in our design. The blog post listing shows titles only, the RSS syndication pushes full articles. So I'm actually not sure why do we have the `<!--more-->` marker there. But I kept it here since we follow it in most of our blog posts. Maybe (long-term) this is something to think about and either use it better in the design or stop using it in the blog post.